library/pdf_creator.class.php

```diff
@@ -0,0 +1,86 @@
+<?php
+/**
+ * PdfCreator class
+ *
+ * @package OpenEMR
+ * @link    http://www.open-emr.org
+ * @author  Jerry Padgett <sjpadgett@gmail.com>
+ * @copyright Copyright (c) 2017 Jerry Padgett <sjpadgett@gmail.com>
+ * @license https://www.gnu.org/licenses/agpl-3.0.en.html GNU Affero General Public License 3
+ */
+use Knp\Snappy\Pdf;
+use Symfony\Component\Debug\ExceptionHandler;
+
+class PdfCreator
+{
+
+    private $binaryPath;
+
+    private $tempPath;
+
+    private function getBinaryPath()
+    {
+        global $webserver_root;
+        // This will not necessarily reflect actual machine bus width but php bus size.
+        $intsize = strlen(decbin(~ 0));
+        if (empty(strstr(php_uname("m"), '64'))) {
+            $bit = "32bit";
+        } else {
+            $bit = "64bit";
+        }
+        try {
```

Comment:
yes but I pulled it because I could not (more likely, would not:)) test. I'll put back and make the pdf stuff a separate PR. I'm going to attempt using sub module to keep the binaries out of main git. I also really think sub module or fit will solve my nagging install issue with my node services. Really glad Robert brought the git management stuff up. Definitely on right track there.
Git is only my 100 and one source control I've had to learn over the years. I just want to code. Oh, and complain, of course :)